test/test_charm.py

```diff
@@ -560,27 +560,42 @@ def _on_foo_bar_action(self, event: ops.ActionEvent):
         charm.on.foo_bar_action.emit(id='1')
 
 
-def test_action_event_defer_fails(
-    request: pytest.FixtureRequest, monkeypatch: pytest.MonkeyPatch, fake_script: FakeScript
+@pytest.mark.parametrize(
+    'event,kwargs',
+    [
+        ('start_action', {'id': 2}),
+        ('stop', {}),
+        ('remove', {}),
+        ('secret_expired', {'id': 'secret:123', 'label': None, 'revision': 0}),
+        ('secret_rotate', {'id': 'secret:234', 'label': 'my-secret'}),
+    ],
+)
+def test_inappropriate_event_defer_fails(
+    event: str,
+    kwargs: typing.Dict[str, typing.Any],
+    request: pytest.FixtureRequest,
+    monkeypatch: pytest.MonkeyPatch,
+    fake_script: FakeScript,
```

Comment:
Minor style nit, somehow I prefer more variable arguments at the end, thus `test_foo(fixtures, parameters)` while it seems that Tony prefers `test_foo(parameters, fixtures)` because then parameter names are closer to the parameter values perhaps?

```suggestion
    request: pytest.FixtureRequest,
    monkeypatch: pytest.MonkeyPatch,
    fake_script: FakeScript,
    event: str,
    kwargs: typing.Dict[str, typing.Any],
```

I don't have a strong opinion either way, it just struck me as something I have not yet seen.

Maybe @benhoyt and @IronCore864 can voice their preference or opinion 🙇🏻

Reply:
I have no real preference here; it just seemed the cleanest at the time. @IronCore864 I assume you must have done a bunch of these in the pytest conversion, but I didn't pay attention to the order when reviewing. I'm happy to go with whatever you did, to keep things most consistent.

Reply:
I believe we've tried to use (fixtures, parameters) in the existing tests, for example https://github.com/canonical/operator/blob/9e5463cde7d942d71cd51af467d832759728480b/test/test_model.py#L2762-L2768